skimage/exposure/_adapthist.py

```diff
@@ -13,21 +13,22 @@
 responsible.  Basically, don't be a jerk, and remember that anything free
 comes with no guarantee.
 """
+from __future__ import division
+import numbers
 import numpy as np
 from .. import img_as_float, img_as_uint
 from ..color.adapt_rgb import adapt_rgb, hsv_value
 from ..exposure import rescale_intensity
-from ..util import view_as_blocks, pad
+from ..util import view_as_windows
+from .._shared.utils import skimage_deprecation
 
 
-MAX_REG_X = 16  # max. # contextual regions in x-direction */
-MAX_REG_Y = 16  # max. # contextual regions in y-direction */
 NR_OF_GREY = 2 ** 14  # number of grayscale levels to use in CLAHE algorithm
 
 
 @adapt_rgb(hsv_value)
-def equalize_adapthist(image, ntiles_x=8, ntiles_y=8, clip_limit=0.01,
-                       nbins=256):
+def equalize_adapthist(image, kernel_size=64, ntiles_x=None, ntiles_y=None,
+                       clip_limit=0.01, nbins=256):
```

Comment:
@blink1073 I'm pretty sure the kernel size kwarg needs to go at the end in order to not break the API. It should also be `None` so that you can check whether it wasn't included, in which case it should be calculated from the default values of `ntiles_`. ie you need to make sure that `equalize_adapthist(im)` returns identical inputs in 0.11 and 0.12, but with a deprecation warning raised.